Documentation/dev-tools/kfence.rst

```diff
@@ -211,26 +224,24 @@ Implementation Details
 ----------------------
 
 Guarded allocations are set up based on the sample interval. After expiration
-of the sample interval, a guarded allocation from the KFENCE object pool is
-returned to the main allocator (SLAB or SLUB). At this point, the timer is
-reset, and the next allocation is set up after the expiration of the interval.
-To "gate" a KFENCE allocation through the main allocator's fast-path without
-overhead, KFENCE relies on static branches via the static keys infrastructure.
-The static branch is toggled to redirect the allocation to KFENCE.
+of the sample interval, the next allocation through the main allocator (SLAB or
+SLUB) returns a guarded allocation from the KFENCE object pool. At this point,
+the timer is reset, and the next allocation is set up after the expiration of
+the interval.  To "gate" a KFENCE allocation through the main allocator's
+fast-path without overhead, KFENCE relies on static branches via the static
+keys infrastructure.  The static branch is toggled to redirect the allocation
+to KFENCE.
 
 KFENCE objects each reside on a dedicated page, at either the left or right
```

Comment:
Are the objects still aligned to `ARCH_SLAB_MINALIGN` when the guard page is on the right side? I'd clarify this here or somewhere.

Reply:
Yes, we honor the slab's requested alignment, and implicitly ARCH_SLAB_MINALIGN (create_cache -> calculate_alignment -> ARCH_SLAB_MINALIGN).

Can add this.